### PR TITLE
 #29 uninitialized variables

### PR DESCRIPTION
**Pull request by sec-matsunaga**
Correct the indication by the static analysis tool.
 Initialization of uninitialized variables.

### CODE REVIEW

File: base_local_planner/src/obstacle_cost_function.cpp [hunk 1]
```diff
@@ -42,10 +42,13 @@
...
 
-ObstacleCostFunction::ObstacleCostFunction(costmap_2d::Costmap2D* costmap) 
-    : costmap_(costmap), sum_scores_(false) {
+ObstacleCostFunction::ObstacleCostFunction(costmap_2d::Costmap2D* costmap)
+    : costmap_(costmap), max_trans_vel_(0.55)
+    , sum_scores_(false), max_scaling_factor_(0.2), scaling_speed_(0.25) {
   if (costmap != NULL) {
     world_model_ = new base_local_planner::CostmapModel(*costmap_);
+  } else {
+    world_model_ = NULL;
```
Comment (r-kurose) [inline base_local_planner/src/obstacle_cost_function.cpp]: It's better to use ```world_model_(NULL)```.

**Review by r-kurose — CHANGES_REQUESTED** [review]

File: base_local_planner/src/obstacle_cost_function.cpp (hunk 1 above)
Comment (sec-matsunaga) [reply base_local_planner/src/obstacle_cost_function.cpp]: Certainly initialize with NULL,
```new base_local_planner :: CostmapModel (* costmap_);```
It is good to set the value with.

**Comment by sec-matsunaga** [comment]
Unnecessary files were being committed.
 from 6c15e6d to 35a583c

File: base_local_planner/src/obstacle_cost_function.cpp (hunk 1 above)
Comment (r-kurose) [reply base_local_planner/src/obstacle_cost_function.cpp]: OK.

**Pull request merged**